### ISSUE TITLE
hhvm core dump Assertion Message: data block = cold

### ISSUE THREAD

**Issue by sandeepraman** [facebook/hhvm#5542]
hhvm is giving a core dump with the following messages:

ThreadType: Web Request
Assertion Message: data block = cold
message: Attempted to emit 1 byte(s) into a 20 byte DataBlock with 0 bytes available. This almost certainly means the TC is full. If this is the case, increasing Eval.JitASize, Eval.JitAColdSize, Eval.JitAFrozenSize and Eval.JitGlobalDataSize in the configuration file when running this script or application should fix this problem.
```
# 0  __restore_rt at sigaction.c:0
# 1  __GI_raise at :0
# 2  __GI_abort at :0
# 3  HPHP::assert_fail(char const*, char const*, unsigned int, char const*, std::basic_string<char, std::char_traits<char>, std::allocator<char> > con
st&) at /usr/local/bin/hhvm:0
# 4  HPHP::jit::translateRegion(HPHP::jit::IRGS&, HPHP::jit::RegionDesc const&, HPHP::hphp_hash_set<HPHP::jit::ProfSrcKey, HPHP::jit::ProfSrcKey::Hash
er, std::equal_to<HPHP::jit::ProfSrcKey> >&, HPHP::jit::TransFlags, std::vector<HPHP::jit::RegionDesc::TypePred, std::allocator<HPHP::jit::RegionDesc:
:TypePred> >&) at /usr/local/bin/hhvm:0
# 5  HPHP::jit::MCGenerator::translateWork(HPHP::jit::TranslArgs const&) at /usr/local/bin/hhvm:0
# 6  HPHP::jit::MCGenerator::translate(HPHP::jit::TranslArgs const&) at /usr/local/bin/hhvm:0
# 7  HPHP::jit::MCGenerator::retranslate(HPHP::jit::TranslArgs const&) at /usr/local/bin/hhvm:0
# 8  HPHP::jit::MCGenerator::createTranslation(HPHP::jit::TranslArgs const&) at /usr/local/bin/hhvm:0
# 9  HPHP::jit::MCGenerator::getCallArrayPrologue(HPHP::Func*) at /usr/local/bin/hhvm:0
# 10 HPHP::jit::funcBodyHelper(HPHP::ActRec*) at /usr/local/bin/hhvm:0
# 11 HPHP::jit::x64::(anonymous namespace)::BackEnd::enterTCHelper(unsigned char*, HPHP::ActRec*) at back-end-x64.cpp:0
# 12 HPHP::jit::MCGenerator::enterTC(unsigned char*, HPHP::ActRec*) at /usr/local/bin/hhvm:0
# 13 HPHP::enterVM(HPHP::ActRec*, HPHP::StackArgsState, HPHP::Resumable*, HPHP::ObjectData*, HPHP::VarEnv*) at bytecode.cpp:0
# 14 HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue*, HPHP::Func const*, HPHP::Variant const&, HPHP::ObjectData*, HPHP::Class*, HPHP::VarEnv*, HPHP::StringData*, HPHP::ExecutionContext::InvokeFlags) at /usr/local/bin/hhvm:0
# 15 HPHP::ExecutionContext::createFunction(HPHP::String const&, HPHP::String const&) at /usr/local/bin/hhvm:0
# 16 HPHP::f_create_function(HPHP::String const&, HPHP::String const&) at /usr/local/bin/hhvm:0
# 17 HPHP::jit::x64::(anonymous namespace)::BackEnd::enterTCHelper(unsigned char*, HPHP::ActRec*) at back-end-x64.cpp:0
# 18 HPHP::jit::MCGenerator::enterTC(unsigned char*, HPHP::ActRec*) at /usr/local/bin/hhvm:0
# 19 HPHP::enterVM(HPHP::ActRec*, HPHP::StackArgsState, HPHP::Resumable*, HPHP::ObjectData*, HPHP::VarEnv*) at bytecode.cpp:0
# 20 HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue*, HPHP::Func const*, HPHP::Variant const&, HPHP::ObjectData*, HPHP::Class*, HPHP::VarEnv*, HPHP::StringData*, HPHP::ExecutionContext::InvokeFlags) at /usr/local/bin/hhvm:0
# 21 HPHP::ExecutionContext::invokeUnit(HPHP::TypedValue*, HPHP::Unit const*) at /usr/local/bin/hhvm:0
# 22 HPHP::include_impl_invoke(HPHP::String const&, bool, char const*) at /usr/local/bin/hhvm:0
# 23 HPHP::hphp_invoke(HPHP::ExecutionContext*, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, HPHP::Array const&, HPHP::VRefParamValue const&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool&, std::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool) at /usr/local/bin/hhvm:0
# 24 HPHP::HttpRequestHandler::executePHPRequest(HPHP::Transport*, HPHP::RequestURI&, HPHP::SourceRootInfo&, bool) at /usr/local/bin/hhvm:0
# 25 HPHP::HttpRequestHandler::handleRequest(HPHP::Transport*) at /usr/local/bin/hhvm:0
# 26 HPHP::ServerWorker<std::shared_ptr<HPHP::FastCGIJob>, HPHP::FastCGITransportTraits>::doJobImpl(std::shared_ptr<HPHP::FastCGIJob>, bool) at /usr/local/bin/hhvm:0
# 27 HPHP::ServerWorker<std::shared_ptr<HPHP::FastCGIJob>, HPHP::FastCGITransportTraits>::doJob(std::shared_ptr<HPHP::FastCGIJob>) at /usr/local/bin/hhvm:0
# 28 HPHP::JobQueueWorker<std::shared_ptr<HPHP::FastCGIJob>, HPHP::Server*, true, false, HPHP::JobQueueDropVMStack>::start() at /usr/local/bin/hhvm:0
# 29 HPHP::AsyncFuncImpl::ThreadFunc(void*) at /usr/local/bin/hhvm:0
# 30 HPHP::start_routine_wrapper(void*) at /usr/local/bin/hhvm:0
# 31 start_thread at pthread_create.c:0
# 32 __clone at /lib64/libc.so.6:0
```
Version: heads/master-0-gae9782928043b091f12547fcf680c81db99f74b9

**Comment by danslo** [comment]
According to https://github.com/facebook/hhvm/issues/4250 , create_function should no longer be leaking since https://github.com/facebook/hhvm/commit/8d0bac44832df988cba9f247beaec5e7041ef6ed

@paulbiss

**Comment by paulbiss** [comment]
Yeah, the fix was incomplete, you'll need to rebase up to 2cbba798bca47c177f1c6a0bf35ce854b82b441a

Sorry about that :0

**Comment by paulbiss** [comment]
FWIW the fix was also cherry-picked into 3.6 and 3.7

**Comment by Orvid** [comment]
Fixed.
